feat: home feed configuration

caaatisgood opened the pull request:
https://github.com/teia-community/teia-ui/issues/301

Make the home feed customizable via `/settings`. See demo vid here https://github.com/teia-community/teia-ui/issues/301#issuecomment-1670335023.

I ended up importing all the feed comps in `home/index` to create a component map. Not the best approach performance-wise but does the job. Happy to do any CRs. (:

caaatisgood committed:
Add home feed configuration

Make home feed customizable via configs https://github.com/teia-community/teia-ui/issues/301

melMass commented:
Thanks! I should be able to review it before next week if no one can earlier, by a quicklook it seems nice, I would just remove the feed names map that is not really needed unless I got the logic wrong

melMass commented on src/constants.ts:
```diff
@@ -68,6 +68,27 @@ export const ALLOWED_FILETYPES_LABEL = Object.entries(MIMETYPE)
   .map((e) => (e[0] === 'ZIP' ? 'HTML (ZIP ARCHIVE)' : e[0]))
   .join(', ')
 
+export const FEED_MAP = {
```
I was referring to this, I don't think it's needed

caaatisgood replied:
This is like a single source of truth shared between the settings page and the home feed component. And also to derive the allowed feed types. I don't know if there's a way to avoid having such a map.

melMass replied:
I simplified it to just a list, I can't fully test it but I just pushed a few edits, I'll be able to properly test it tomorrow and merge then! Thanks again

caaatisgood replied:
Thanks @melMass!

melMass committed:
chore 🧹: use ts for home index

melMass committed:
refactor 🔨: use array instead of map

We don't really need a map, also simplifies the type constraints, added extra eslint rules for ts

melMass committed:
style 💎: strip emojis from feed in settings

it seems there is no easy way to useTwemoji in the ReactSelect options

melMass committed:
Merge branch 'main' into feature/home-feed-configuration-301
